odoo/addons/base/models/ir_attachment.py

```diff
@@ -343,15 +343,14 @@ def _check_serving_attachments(self):
 
     @api.model
     def check(self, mode, values=None):
-        """Restricts the access to an ir.attachment, according to referred model
-        In the 'document' module, it is overridden to relax this hard rule, since
-        more complex ones apply there.
-        """
+        """ Restricts the access to an ir.attachment, according to referred mode """
         if self.env.is_superuser():
             return True
+        # Always require an internal user (aka, employee) to access to a attachment
+        if not (self.env.is_admin() or self.env.user.has_group('base.group_user')):
```

Comment:
I'm not sure to understand why this condition on user is necessary since https://github.com/odoo/odoo/commit/62c9dedafda0cbdc618641eaa80c49535ab0b5f6#diff-2ad4044a5fe3591bd1630b6e94057ae3R3 set access 0,0,0,0 for public users already.

What do I miss?

Reply:
Super isn't called, and the check function is call in multiple place by other model. Then I think it is legit.